### ISSUE TITLE
Dropped tables can not be physically removed if `syncAllSchema` is trigger frequently

### ISSUE THREAD

**Issue by JaySon-Huang** [pingcap/tiflash#9227]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Run the following SQLs repeatedly

```
CREATE TABLE if not exists test_range_1 (id INT,name VARCHAR(20))
PARTITION BY RANGE (id) (PARTITION p0 VALUES LESS THAN (10),PARTITION p1 VALUES LESS THAN (20),PARTITION p2 VALUES LESS THAN (30),PARTITION p3 VALUES LESS THAN (40),PARTITION p4 VALUES LESS THAN (50);

CREATE TABLE if not exists test_swap_1 (id INT,name VARCHAR(20));
ALTER TABLE test_range_1 EXCHANGE PARTITION p0 WITH TABLE test_swap_1;
DROP TABLE test_swap_1;
```

### 2. What did you expect to see? (Required)

The dropped table can be successfully removed from tiflash instances

### 3. What did you see instead (Required)

The dropped tables' IStorage instances remain in TiFlash even after the gc_safepoint is exceed than the drop timepoint + gc_lifetime.
More and more ".sql" files of the dropped tables' remain in TiFlash instance disk. TiFlash sync all schema become more and more slower.

### 4. What is your TiFlash version? (Required)
v6.5.10

**Comment by JaySon-Huang** [comment]
In the previous versions

1. Repeatedly execute the given SQLs. The rapidly swap and dropped table_id will lead to some warnings like "miss table" or "miss partition table in TiKV" with "Errors::DDL::StaleSchema"
https://github.com/pingcap/tiflash/blob/28a2314dd36fc1e990b9d8a3a6aec89ba8a220b8/dbms/src/TiDB/Schema/SchemaBuilder.cpp#L871-L877


2. When tiflash meet exception in `tryLoadSchemaDiffs`, tiflash will fallback to `loadAllSchema` to fetch all tables info from TiKV.
https://github.com/pingcap/tiflash/blob/28a2314dd36fc1e990b9d8a3a6aec89ba8a220b8/dbms/src/TiDB/Schema/TiDBSchemaSyncer.h#L159-L164

3. In `SchemaBuilder::syncAllSchema`, each table that does not exists in tikv but still in the tiflash instance. TiFlash will execute `applyDropPhysicalTable` for each table, wich will update the "tombstone_ts" of the table in its ".sql" file.
Even if the table is already "tombstone", the function will update the "tombstone_ts" by a newer tso from PD. This make the IStorage instance still exist even after the gc_safepoint is exceed than the drop timepoint + gc_lifetime.
https://github.com/pingcap/tiflash/blob/28a2314dd36fc1e990b9d8a3a6aec89ba8a220b8/dbms/src/TiDB/Schema/SchemaBuilder.cpp#L1437-L1451
https://github.com/pingcap/tiflash/blob/28a2314dd36fc1e990b9d8a3a6aec89ba8a220b8/dbms/src/TiDB/Schema/SchemaBuilder.cpp#L1240-L1268

**Comment by JaySon-Huang** [comment]
Close as it will be fixed in the next patch version of release-6.5 and release-7.1. v7.5 or later are not affected.
